cloud-platform/add-secrets-to-deployment.md

```diff
@@ -0,0 +1,170 @@
+* You have previously deployed your application. See [Deploying-an-application-to-the-Cloud-Platform](https://github.com/ministryofjustice/cloud-platform-user-docs/blob/master/cloud-platform/app-deploy.md/#deploying-an-application-to-the-cloud-platform)
+* Check your deployment is running. See [Interacting-with-the-application](https://github.com/ministryofjustice/cloud-platform-user-docs/blob/master/cloud-platform/app-deploy.md/#interacting-with-the-application)
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+See [kuberenetes-using-secrets-as-environment-variables](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables)
+
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+put the aws key secret and id into a file (in this example demo-v9.k8s) as follows ('thekey' referencing the aws_access_key_id
+and 'thesecret' referencing the aws_secret_access_key):
+
+```
+apiVersion: v1
+
+kind: Secret
+
+metadata:
+
+  name: demosecret-v9
+
+type: Opaque
+
+data:
+
+  thekey: AKIAFTKSAW15HJLOGD
+
+  thesecret: g8hjpmhvgfhk4547gfdshhjj
+
+```
+
+create secret generic secret from that file:
+```$kubectl -n demo-app create secret generic demosecret-v9 --from-file=demosecret-v9=demo-v9.k8s```
+
+'demo-app' being the namespace, 'demosecret-v9' referencing the 'name' in demo-v9.k8s and referencing the file 'demo-v9.k8s'
+this will give the following screen output
+
+
+```
+secret "demosecret-v9" created
+```
+To view the output of your secret:
+
+```
+$kubectl -o json -n demo-app get secret demosecret-v9
+
+```
+
+This will output something like the following:
+
+```
+{
+    "apiVersion": "v1",
+    "data": {
+        "demosecret-v9": "YXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGRlbW9zZWNyZXQtdjkKdHlwZTogT3BhcXVlCmRhdGE6CiAgdGhla2V5OiBBS0lBRlRLU0FXMTVISkxPR0QKICB0aGVzZWNyZXQ6IGc4aGpwbWh2Z2ZoazQ1NDdnZmRzaGhqago="
+    },
+    "kind": "Secret",
+    "metadata": {
+        "creationTimestamp": "2018-05-09T15:17:50Z",
+        "name": "demosecret-v9",
+        "namespace": "demo-app",
+        "resourceVersion": "3021923",
+        "selfLink": "/api/v1/namespaces/demo-app/secrets/demosecret-v9",
+        "uid": "2315be2f-539c-11e8-8ff4-0aecd97fec6e"
+    },
+    "type": "Opaque"
+}
+
+```
+Using the 'data' output from above, issue the following command (to validate and verify that it is the same as you originally inputted into the demo-v9.k8s file above):
+
+```
+$ base64 -D
+YXBpVmVyc2lvbjogdjEKa2luZDogU2VjcmV0Cm1ldGFkYXRhOgogIG5hbWU6IGRlbW9zZWNyZXQtdjkKdHlwZTogT3BhcXVlCmRhdGE6CiAgdGhla2V5OiBBS0lBRlRLU0FXMTVISkxPR0QKICB0aGVzZWNyZXQ6IGc4aGpwbWh2Z2ZoazQ1NDdnZmRzaGhqago=
+
+```
+ctrl+d to exit
```

Comment:
This feels like a typo? The `ctrl+d to exit` doesn't seem to fit with the rest of the flow and would kill the shell if you did it.